### PR TITLE
[minor] give a name to the transport in the Worker

### PR DESCRIPTION
**Pull request by nikophil**
Hello!

I hope this was not made on purpose, but in order to best mimic the behavior of messenger, the receivers need to have a name in the worker, as seen in [`Symfony\Component\Messenger\Command\ConsumeMessagesCommand`](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php#L216)

```php
// Symfony\Component\Messenger\Command\ConsumeMessagesCommand
protected function execute(InputInterface $input, OutputInterface $output): int
{
        $receivers = [];
        foreach ($receiverNames = $input->getArgument('receivers') as $receiverName) {
            //...
            $receivers[$receiverName] = $this->receiverLocator->get($receiverName);
        }

        // ...
        $worker = new Worker($receivers, $bus, $this->eventDispatcher, $this->logger);
        // ...
}
```

Otherwise, we cannot reproduce the retry behavior because the [computation of `retryStrategy` is based on transport's name](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php#L52). Yet, IMO, it could be a good thing to be able to reproduce retry behavior in tests since I've already had some complex bugs in retry scenario.

### CODE REVIEW

File: tests/InteractsWithMessengerTest.php [hunk 1]
```diff
@@ -390,7 +390,6 @@ public function can_access_sent_acknowledged_and_rejected(): void
         $this->assertCount(0, $this->messenger()->rejected());
 
         $this->messenger()->process();
-
```
Comment (kbond) [inline tests/InteractsWithMessengerTest.php]: Can you remove this unrelated change?

File: tests/Fixture/config/multi_transport.yaml [hunk 2]
```diff
@@ -4,8 +4,14 @@ imports:
 framework:
     messenger:
         transports:
-            async1: test://
-            async2: test://?intercept=false&catch_exceptions=false
+            async1:
+                dsn: test://
+                retry_strategy:
+                    max_retries: 0
```
Comment (kbond) [inline tests/Fixture/config/multi_transport.yaml:10]: What is the purpose of having the `retry_strategy `set here if we aren't testing it? Should we be explicitly testing it?

**Review by kbond — CHANGES_REQUESTED** [review]
(Rebase needed)

It was unintentional to leave out the name so this is a valid fix.

Is this related to #2? Should/could we add tests to ensure failed messages are added to a failed transport. I'm not saying this should be done here in this PR, I'm curious for your opinion.

File: tests/Fixture/config/multi_transport.yaml (hunk 2 above)
Comment (nikophil) [reply tests/Fixture/config/multi_transport.yaml:10]: Actually the default retry strategy is: 
```php
new MultiplierRetryStrategy(
    maxRetries: 3,
    delayMilliseconds: 1000,
    multiplier: 1,
    maxDelayMilliseconds: 0
);
```

but it wasn't working before this PR because the `$retryStrategyLocator` in `SendFailedMessageForRetryListener` was failing to find it. Now it works, then the default retry strategy is used and the test which are testing failures dispatch more messages than expected.

**Comment by nikophil** [comment]
I've rebased the PR.

> Is this related to #2? 

I think this isn't related to #2. Failure transport is only related to retry if the event is `setForRetry`

> Should/could we add tests to ensure failed messages are added to a failed transport.

I think we should not behave as if it does not exist, and then at least configure `max_retries: 1` for one transport and test that in blocking and unblocking modes a failing message is dispatched twice

File: tests/Fixture/config/multi_transport.yaml (hunk 2 above)
Comment (kbond) [reply tests/Fixture/config/multi_transport.yaml:10]: Got it.

**Review by kbond — APPROVED** [review]
Perfect, thanks for the explanation!
